roles/matrix-bot-maubot/tasks/setup_install.yml

```
@@ -0,0 +1,113 @@
+---
+
+- name: Ensure maubot paths exist
+  file:
+    path: "{{ item.path }}"
+    state: directory
+    mode: 0755
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+  with_items:
+    - {path: "{{ matrix_bot_maubot_base_path }}", when: true}
+    - {path: "{{ matrix_bot_maubot_data_path }}", when: true}
+    - {path: "{{ matrix_bot_maubot_docker_src_files_path }}", when: true}
+  when: "item.when|bool"
+
+- name: Ensure maubot configuration file created
+  template:
+    src: "{{ role_path }}/templates/config/config.yaml.j2"
+    dest: "{{ matrix_bot_maubot_data_path }}/config.yaml"
+    owner: "{{ matrix_user_username }}"
+    group: "{{ matrix_user_groupname }}"
+    mode: "u=rwx"
+
+- name: Generate Maubot proxying configuration for matrix-nginx-proxy
+  set_fact:
+    matrix_bot_maubot_matrix_nginx_proxy_configuration: |
+      location ~ ^/(_matrix/maubot/.*) {
+      {% if matrix_nginx_proxy_enabled|default(False) %}
+      	{# Use the embedded DNS resolver in Docker containers to discover the service #}
+      	resolver 127.0.0.11 valid=5s;
+      	set $backend "matrix-maubot:{{ matrix_bot_maubot_port }}/$1";
```

The port inside the container is always `29316`, isn't it? This probably doesn't need to use this variable.

---

Yeah it is. I thought this port refers to the exposed port which could be changed -> therefore the variable

---

It seems useful in the `{% else %}` clause. In there, it refers to the exposed port (on `127.0.0.1`).

The main `{% if %}` part (before `{% else %}`) handles proxying over the container network. There, `matrix-bot-maubot:29316` is always true.

By the way, there's a typo here. Now that the container is named `matrix-bot-maubot` (not `matrix-maubot`), this line needs to be fixed up:

```suggestion
      	set $backend "matrix-bot-maubot:29316/$1";
```

---

By the way, it seems like this `set $backend` line (and likely the surrounding lines too) mix different ways indenting - tabs and spaces.

We should switch to **spaces-only** here, because YAML doesn't like tabs.

---

I fixed this in my PR to moan0s' branch.

---

done